feat: upgrade project to react-native 0.81.0 and react 19

ctriantaf opened the pull request:
# Overview
- Upgrade library to use react-native 0.81.1 and react 19.1.0
- Fix new arch implementation
- Upgrade example to use create-react-native-library and link local project

## The following were tested:
- New Arch
- Old Arch
- Expo ~54.0.10

## Notes
- Copilot tests were removed because IMO they didn't make much sense
- Expo setup for `Bridging-Header.h` on iOS and `AppDelegate.swift` were removed per this [comment](https://github.com/kesha-antonov/react-native-background-downloader/issues/97#issuecomment-3381350782). Please advice on this.
  - [Commit #1](https://github.com/kesha-antonov/react-native-background-downloader/commit/7100a892cb0c0e9aff9ff19b5ebc6280dead5a04)
  - [Commit #2](https://github.com/kesha-antonov/react-native-background-downloader/commit/803475cd266a807b4562e97da19f097d7e2a126f)

ctriantaf committed:
feat: upgrade project to react-native 0.81.0
- support new architecture
- upgrade example to use create-react-native-library

ctriantaf committed:
chore: fix ios crash when initializing

ctriantaf committed:
chore: add Podfile.lock

ctriantaf committed:
chore: remove example/vendor

ctriantaf committed:
chore: add lib/ folder (as it won't be released to npm)

ctriantaf committed:
chore: delete plugin/

ctriantaf committed:
Revert "chore: delete plugin/"

This reverts commit 85a3b5f8e768ef368f5eec5754553843140fce3c.

ctriantaf committed:
chore: update expo plugin, add @expo/config-plugins as dev dependency

ctriantaf committed:
chore: remove ios Bridging-Header setup per comment issues/97#issuecomment-3381350782

ctriantaf committed:
chore: remove AppDelegate code per comment issues/97#issuecomment-3381350782

ctriantaf committed:
chore: add support for old arch in ios

ctriantaf committed:
chore: fix ios scheme

kesha-antonov commented:
Thanks!

Could you please add `example/vendor` to gitignore  and make `git rm -rf` on this dir to remove it?
It will be generated on `bundle install` so no need to put it in project

ctriantaf committed:
chore: add example.scheme for ios

ctriantaf committed:
chore: add example/vendor to gitignore

ctriantaf commented:
@kesha-antonov the `vendor/` is tracked in the `main` branch of the library but I have removed in my PR.

Although the `example/vendor/` directory was not created for me during `bundle install` probably due to my bundle settings, I've added it to `gitignore`.

kesha-antonov commented:
I updated the code directly in `main`
Updated to latest RN lib's code & example
Because it's hard to review such a large PR, it's easier to make smaller PRs for different tasks

I'll get back later when I'll test updated code

brightsider commented:
ErrorHandlerObject and ProgressHandlerObject couldn't be imported in this fork

brightsider commented:
Also, fatal error: 'RNBackgroundDownloader.h' file not found, so can't use it in handleEventsForBackgroundURLSession

kesha-antonov commented:
Thanks for the PR!

Fixed in main
